### PR TITLE
fix(administration) : Make the users not created on centreon plateform unenabled to edit their profile information 

### PR DESCRIPTION
**Pull request by Exodiaaaa**
## Description

Users created not using Centreon Plateform cannot update their information(Alias, Name, Email, Password Management)

**Fixes** # (MON-24798)

## Type of change

- [X] Patch fixing an issue (non-breaking change)
- [ ] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software

## Target serie

- [ ] 22.04.x
- [ ] 22.10.x
- [ ] 23.04.x
- [ ] 23.10.x
- [X] 24.04.x (master)

<h2> How this pull request can be tested ? </h2>

Given a user authenticated through SSO (or LDAP) and having rights to access to “Administration > My Account”
When he access to “Administration > My Account” menu
He can’t change his following properties:

- Name
- Alias / Login
- Email

#### Community contributors & Centreon team

- [X] I have followed the **coding style guidelines** provided by Centreon
- [ ] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [ ] I have commented my code, especially **hard-to-understand areas** of the PR.
- [X] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

File: centreon/www/include/Administration/myAccount/formMyAccount.ihtml [hunk 1]
```diff
@@ -38,7 +38,7 @@
     		<tr class="list_two"><td class="FormRowField">{$form.contact_pager.label}</td><td class="FormRowValue">{$form.contact_pager.html}</td></tr>
             <tr class="list_one"><td class="FormRowField">{$form.contact_lang.label}</td><td class="FormRowValue">{$form.contact_lang.html}</td></tr>
             <tr class="list_two"><td class="FormRowField">{$form.contact_location.label}</td><td class="FormRowValue">{$form.contact_location.html}</td></tr>
-            {if $cct.contact_auth_type != 'ldap'}
+            {if $cct.contact_auth_type === 'local'}
```
Comment (jeremyjaouen) [inline centreon/www/include/Administration/myAccount/formMyAccount.ihtml:41]: Is this working for user connecting through openid ?

File: centreon/www/include/Administration/myAccount/formMyAccount.php [hunk 2]
```diff
@@ -106,14 +106,18 @@
 $form = new HTML_QuickFormCustom('Form', 'post', "?p=" . $p);
 $form->addElement('header', 'title', _("Change my settings"));
 $form->addElement('header', 'information', _("General Information"));
-$form->addElement('text', 'contact_name', _("Name"), $attrsText);
-if ($cct["contact_auth_type"] != 'ldap') {
+if ($cct["contact_auth_type"] === 'local') {
```
Comment (jeremyjaouen) [inline centreon/www/include/Administration/myAccount/formMyAccount.php:109]: Is this working for user connecting through openid ?

Comment (Exodiaaaa) [reply centreon/www/include/Administration/myAccount/formMyAccount.php:109]: its working only for user with contact_auth_type = local for otheir using ldap or SSO the alias, Name, Email, Password Management theyre not allowed to it

**Review by callapa — CHANGES_REQUESTED** [review]
Just because the form is blocked doesn't mean the user can't send the information.

**Comment by Exodiaaaa** [comment]
that's what is demanded on the ticket is to freeze the input

**Review by jeremyjaouen — DISMISSED** [review]

File: centreon/www/include/Administration/myAccount/formMyAccount.php [hunk 3]
```diff
@@ -414,14 +418,25 @@ function myReplace()
...
+if ($cct['contact_auth_type'] !== 'local') {
+    if ($_POST['contact_name'] !== $cct['contact_name'] || empty($_POST['contact_name'])) {
+        $_POST['contact_name'] = $cct['contact_name'];
+    }
+    if ($_POST['contact_alias'] !== $cct['contact_alias'] || empty($_POST['contact_alias'])) {
+        $_POST['contact_alias'] = $cct['contact_alias'];
+    }
+    if ($_POST['contact_email'] !== $cct['contact_email'] || empty($_POST['contact_email'])) {
+        $_POST['contact_email'] = $cct['contact_email'];
+    }
```
Comment (callapa) [inline centreon/www/include/Administration/myAccount/formMyAccount.php:430]: Simplify:
```suggestion
    $_POST['contact_name'] = $cct['contact_name'];
    $_POST['contact_alias'] = $cct['contact_alias'];
    $_POST['contact_email'] = $cct['contact_email'];
```

**Pull request closed without merging**